src/hints/syscalls.rs

```diff
@@ -352,6 +356,116 @@ pub fn set_syscall_ptr(
+            builtin_ptrs=ids.builtin_ptrs,
+            range_check_ptr=ids.range_check_ptr,
+            selector=selector,
+        )"#
+};
+pub fn os_logger_enter_syscall_preprare_exit_syscall(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    println!("TODO: os_logger enter/exit calls");
+
+    Ok(())
+}
+
+pub const EXIT_SYSCALL: &str = indoc! {r#"
+    exit_syscall(selector=ids.CALL_CONTRACT_SELECTOR)"#
+};
+pub fn exit_syscall(
+    _vm: &mut VirtualMachine,
+    _exec_scopes: &mut ExecutionScopes,
+    _ids_data: &HashMap<String, HintReference>,
+    _ap_tracking: &ApTracking,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    println!("exit_syscall hint: TODO");
+
+    Ok(())
+}
+
+pub const FETCH_STATE_ENTRY_5: &str = indoc! {r#"
+	# Fetch a state_entry in this hint and validate it in the update that comes next.
+	ids.state_entry = __dict_manager.get_dict(ids.contract_state_changes)[
+	    ids.contract_address
+	]
+
+	ids.new_state_entry = segments.add()"#
+};
+pub fn fetch_state_entry_5(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let key = get_integer_from_var_name("contract_address", vm, ids_data, ap_tracking)?;
+    let dict_ptr = get_ptr_from_var_name("contract_state_changes", vm, ids_data, ap_tracking)?;
+    let val = match exec_scopes.get_dict_manager()?.borrow().get_tracker(dict_ptr)?.data.clone() {
+        Dictionary::SimpleDictionary(dict) => dict
+            .get(&MaybeRelocatable::Int(key.into_owned()))
+            .expect("State changes dictionnary shouldn't be None")
+            .clone(),
+        Dictionary::DefaultDictionary { dict: _d, default_value: _v } => {
+            panic!("State changes dict shouldn't be a default dict")
+        }
+    };
+    insert_value_from_var_name("state_entry", val, vm, ids_data, ap_tracking)?;
+    insert_value_from_var_name("new_state_entry", vm.add_memory_segment(), vm, ids_data, ap_tracking)?;
+
+    Ok(())
+}
+
+pub const CACHE_CONTRACT_STORAGE_2: &str = indoc! {r#"
+	# Make sure the value is cached (by reading it), to be used later on for the
+	# commitment computation.
+	value = execution_helper.storage_by_address[ids.contract_address].read(
+	    key=ids.syscall_ptr.request.address
+	)
+	assert ids.value == value, "Inconsistent storage value.""#
+};
+pub fn cache_contract_storage_2(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let key = get_integer_from_var_name("request_address", vm, ids_data, ap_tracking)?;
```

Comment:
This line is wrong as the hint wants to use `ids.syscall_ptr.request.address` (cc @maciejka )